src/bacnet/basic/object/bv.c

```diff
@@ -1367,6 +1367,90 @@ static struct object_data *Binary_Value_Object_Index(int index)
 {
     return Keylist_Data_Index(Object_List, index);
 }
+
+/**
+ * For a given object instance-number, returns the event_enable property value
+ *
+ * @param  object_instance - object-instance number of the object
+ *
+ * @return  event_enable property value
+ */
+uint32_t Binary_Value_Event_Enable(uint32_t object_instance)
+{
+    uint32_t event_enable = 0;
+    struct object_data *pObject = Binary_Value_Object(object_instance);
+
+    if (pObject) {
+        event_enable = pObject->Event_Enable;
+    }
+
+    return event_enable;
+}
+
+/**
+ * For a given object instance-number, sets the event_enable property value
+ *
+ * @param object_instance - object-instance number of the object
+ * @param event_enable - event_enable property value
+ *
+ * @return true if the event_enable property value was set
+ */
+bool Binary_Value_Event_Enable_Set(uint32_t object_instance, uint32_t event_enable)
+{
+    bool status = false;
+    struct object_data *pObject = Binary_Value_Object(object_instance);
+
+    if (pObject) {
+      if(!(event_enable & ~(EVENT_ENABLE_TO_OFFNORMAL | EVENT_ENABLE_TO_FAULT | EVENT_ENABLE_TO_NORMAL))) {
```

Comment:
Could you add a @note in the header as to why this test is there ? Just reading the header it seems we set the event enable for a given instance regardless... without a test